### ISSUE TITLE
Scores dimensions of blip2_t5 outputs

### ISSUE THREAD

**Issue by aburns4** [salesforce/LAVIS#343]
Hi! I want to access the logits over the entire vocabulary at each decoding step from a blip2 t5 model. I modified the generate function to pass `output_scores` and `return_dict_in_generate` (both set to `True`) to allow me to access the `sequences` and `scores` available per https://huggingface.co/docs/transformers/internal/generation_utils#transformers.generation.BeamSearchEncoderDecoderOutput

However, I am finding the length of the scores tuple returned is larger than the decoded sequence length. From my understanding `len(scores) = sequence_length - 1` due to the BOS token. Yet, I am getting `len(scores) = 6` or 8 or other values > `sequence_length` but < `max_length` for a few examples I've looked at with sequences with decoded length 3. 

Trying `compute_transition_scores` per https://huggingface.co/docs/transformers/main/en/main_classes/text_generation#transformers.GenerationMixin.compute_transition_scores also fails due to this dimension mismatch.

Any idea what could be going on?

Thank you very much!

**Comment by aburns4** [comment]
Realized I just needed to pass in the beam indices. Closing.
